docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/waiting/ClusterHealthCheck.java

```diff
@@ -37,5 +43,26 @@ static ClusterHealthCheck serviceHealthCheck(String containerName, HealthCheck<C
         };
     }
 
-    SuccessOrFailure isClusterHealthy(Cluster cluster);
+    static ClusterHealthCheck nativeHealthChecks() {
+        return cluster -> {
+            Set<String> unhealthyContainers = new LinkedHashSet<>();
+            try {
+                for (Container container : cluster.allContainers()) {
+                    State state = container.state();
+                    if (state == State.UNHEALTHY) {
```

Comment:
Perhaps worth capturing in a javadoc comment that this will not wait for any DOWN or PAUSED containers.